web/static/js/phoenix_html.js

```diff
@@ -1,38 +1,40 @@
-function isLinkToSubmitParent(element) {
-  var isLinkTag = element.tagName === 'A';
-  var shouldSubmitParent = element.getAttribute('data-submit') === 'parent';
-
-  return isLinkTag && shouldSubmitParent;
-}
+(function() {
+  function buildHiddenInput(name, value) {
+    var input = document.createElement("input");
+    input.type = "hidden";
+    input.name = name;
+    input.value = value;
+    return input;
+  }
 
-function getClosestForm(element) {
-  while (element && element !== document && element.nodeType === Node.ELEMENT_NODE) {
-    if (element.tagName === 'FORM') {
-      return element;
+  function handleLinkClick(link) {
+    if(link.dataset["confirm"] == "true" && !window.confirm("Are you sure?")) {
+        return;
     }
-    element = element.parentNode;
+
+    var to = link.dataset["to"],
+        method = buildHiddenInput("_method", link.dataset["method"]),
+        csrf = buildHiddenInput("_csrf_token", link.dataset["csrf"]),
+        form = document.createElement("form");
+
+    form.method = "post";
+    form.action = to;
+    form.style.display = "hidden";
+
+    form.appendChild(csrf);
+    form.appendChild(method);
+    document.body.appendChild(form);
+    form.submit();
   }
-  return null;
-}
 
-function didHandleSubmitLinkClick(element) {
-  while (element && element.getAttribute) {
-    if (isLinkToSubmitParent(element)) {
-      var message = element.getAttribute('data-confirm');
-      if (message === null || confirm(message)) {
-        getClosestForm(element).submit();
+  window.onload = function() {
```

Comment:
You're correct! I'll dig back into this after work today. If you see anything else let me know.